### PR TITLE
fix 2 bugs. see #12 and #18.

### PR DESCRIPTION
**Pull request by ttyniwa**
I found 2 bugs.

bug1.
  Mention count does not disappear on active CHANNEL.
  
bug2.
  Mention count and Unread badge does not appear on active DM.

Tested OS: OSX El Capitan
Tested Mattermost version: 1.4
Test case: 
Post to channel/DM when window isn't focused, unread badge(if DM, count badge) appear. and dissapear when it focused.
Post to channel/DM with mention when window isnt focused, count badge appear. and dissapear when it focused.

### CODE REVIEW

**Comment by mid0111** [comment]
Thank you for fixing the bug. I forgot to check about DM.

Then, I met the following behavior when I test this PR.
* Post to active channel/DM with mention when window isn't focused, I expect __count badge__ appear but __unread badge__ appear.

File: src/browser/webview/mattermost.js [hunk 1]
```diff
@@ -89,8 +72,57 @@ function overrideNotification() {
...
-      ipc.sendToHost('onActiveChannelNotify');
+
+    // mentionCount for active channel
+    var newSeparators = document.getElementsByClassName('new-separator');
+    var post;
+    var isMentioned = false;
+    // Skip until real new-separator appear.
+    for (var i = 0; i < newSeparators.length; i++) {
+      if (newSeparators[i].offsetParent !== null) {
+        post = newSeparators[i];
```
Comment (yuya-oc) [inline src/browser/webview/mattermost.js]: On Notification, `post` may be `null` depending on its timing on my environment (Mattermost 1.1). So I think it's better to manage only `notified` flag here, count mentions in interval timer.

**Comment by ttyniwa** [comment]
* I fixed bug and refactored.
* I tested all case i realized.
* I'll eat my own dog food tomorrow.

**Comment by ttyniwa** [comment]
It was working correctly today.

**Comment by yuya-oc** [comment]
Thanks. Sorry, I can't test it until Saturday. Please wait.

**Comment by yuya-oc** [comment]
I think it works correctly including unfocused tabs.

Just for my check, please tell me whether following points are correct.
* App doesn't count sidebar-badges.
* In active channel, mentions are counted as '1' regardless of the number of its highlighted posts.

**Comment by ttyniwa** [comment]
>    App doesn't count sidebar-badges.
>    In active channel, mentions are counted as '1' regardless of the number of its highlighted posts.

Sorry, I misunderstood slack app behavior.
fixed it.

**Comment by yuya-oc** [comment]
I created a topic branch `pr/22` to fix 2 bugs. Please check.

1. badge-notify exists on navbar-toggle in mobile layout.
   It has also badge class, so mention count becames twice.
twiced mention count


2. [] accessor was missing. Badge count becomes unexpected string.
unexpected string

**Comment by ttyniwa** [comment]
@yuya-oc Thank you for fixing bugs. I think it's ok!

**Comment by yuya-oc** [comment]
Thank you very much. I will merge.

**Pull request merged**